integration/flink-native-listener/src/main/java/io/openlineage/flink/visitor/KafkaTableLineageDatasetIdentifierVisitor.java

```diff
@@ -0,0 +1,72 @@
+/*
+/* Copyright 2018-2024 contributors to the OpenLineage project
+/* SPDX-License-Identifier: Apache-2.0
+*/
+
+package io.openlineage.flink.visitor;
+
+import io.openlineage.client.utils.DatasetIdentifier;
+import io.openlineage.client.utils.DatasetIdentifier.Symlink;
+import io.openlineage.client.utils.DatasetIdentifier.SymlinkType;
+import io.openlineage.flink.wrapper.TableLineageDatasetImplWrapper;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.Optional;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.flink.streaming.api.lineage.LineageDataset;
+import org.apache.flink.table.catalog.CatalogBaseTable;
+import org.apache.flink.table.planner.lineage.TableLineageDatasetImpl;
+
+/** Class to extract dataset identifier from {@link TableLineageDatasetImpl} stored on Kafka. */
+@Slf4j
+public class KafkaTableLineageDatasetIdentifierVisitor implements DatasetIdentifierVisitor {
+  private static final String KAFKA_CONNECTOR = "kafka";
+  private static final String KAFKA_DATASET_PREFIX = "kafka://";
+  private static final String COMMA = ",";
+  private static final String SEMICOLON = ";";
+
+  @Override
+  public boolean isDefinedAt(LineageDataset dataset) {
+    TableLineageDatasetImplWrapper implWrapper = new TableLineageDatasetImplWrapper(dataset);
+
+    Optional<CatalogBaseTable> table = implWrapper.getCatalogBaseTable(dataset);
+    if (!table.isPresent()) {
+      return false;
+    }
+
+    if (table.get().getOptions() == null) {
+      return false;
+    }
+
+    return table.get().getOptions().containsKey("connector")
+        && KAFKA_CONNECTOR.equals(table.get().getOptions().get("connector"));
```

Comment:
An example SQL can be helpful in understanding this:
```
 String.format(
              "create table kafka_input (\n"
                  + "  `computed-price` as price + 1.0,\n"
                  + "  price decimal(38, 18),\n"
                  + "  currency string,\n"
                  + "  log_date date,\n"
                  + "  log_time time(3),\n"
                  + "  log_ts timestamp(3),\n"
                  + "  ts as log_ts + INTERVAL '1' SECOND,\n"
                  + "  watermark for ts as ts\n"
                  + ") with (\n"
                  + "  'connector' = '%s',\n"
                  + "  'topic' = '%s',\n"
                  + "  'properties.bootstrap.servers' = '%s',\n"
                  + "  'properties.group.id' = '%s',\n"
                  + "  'scan.startup.mode' = 'earliest-offset',\n"
                  + "  'format' = 'json'\n"
                  + ")",
```
Looking at the table properties is required. Once we realise a connector is Kafka, we can extract dataset identifier from bootstrap servers and topic name. Unfortunately, for SQL we need to understand the connector to know which properties to use. Or at least, I am not aware of workaround to avoid this.

Reply:
I think topic name is the correct dataset identifier. However, we should also include table name as symlink. It's one of the things I would recommend doing in separate PRs after this one is merged. I put at the end of `README.md` list of know issues and future enhancements and put Kafka table symlink on that list.